lib/PuppeteerSharp/Helper.cs

```diff
@@ -19,7 +20,38 @@ internal static object ValueFromRemoteObject(dynamic remoteObject)
                         throw new Exception("Unsupported unserializable value: " + remoteObject.unserializableValue);
                 }
             }
-            return remoteObject.value;
+
+            if (remoteObject.value == null)
+            {
+                return null;
+            }
+
+            // https://chromedevtools.github.io/devtools-protocol/tot/Runtime#type-RemoteObject
+            string objectValue = remoteObject.value.ToString();
+            string objectType = remoteObject.type.ToString();
+            
+            switch (objectType)
+            {
+                case "object":
+                    return JsonConvert.DeserializeObject<T>(objectValue);
```

Comment:
I think JsonConvert will be able to serialize a "number" a "boolean" and a "bigint", what do you think?

Reply:
Yes it's pretty sure it can